src/libraries/System.Linq/tests/AnyTests.cs

```diff
@@ -20,6 +20,38 @@ public void SameResultsRepeatCallsIntQuery()
             Assert.Equal(q.Any(predicate), q.Any(predicate));
         }
 
+        [Fact]
+        public void SameResultsRepeatCallsIntArray() {
+            var array = new[] { 9999, 0, 888, -1, 66, -777, 1, 2, -12345 };
+
+            Func<int, bool> predicate = IsEven;
+            Assert.Equal(array.All(predicate), array.All(predicate));
+        }
+
+        [Fact]
+        public void SameResultsRepeatCallsListOfInt() {
+            var list = new List<int>() { 9999, 0, 888, -1, 66, -777, 1, 2, -12345 };
+
+            Func<int, bool> predicate = IsEven;
+            Assert.Equal(list.All(predicate), list.All(predicate));
+        }
+
+        [Fact]
+        public void SameResultsRepeatCallsIntArray() {
+            var array = new[] { 9999, 0, 888, -1, 66, -777, 1, 2, -12345 };
+
+            Func<int, bool> predicate = IsEven;
+            Assert.Equal(array.Any(predicate), array.Any(predicate));
+        }
+
+        [Fact]
+        public void SameResultsRepeatCallsListOfInt() {
+            var list = new List<int> { 9999, 0, 888, -1, 66, -777, 1, 2, -12345 };
+
+            Func<int, bool> predicate = IsEven;
+            Assert.Equal(list.Any(predicate), list.Any(predicate));
+        }
+
```

Comment:
You've missed the point of these tests - it's not about `Any()`/`All()`; instead, it's about the LINQ query.